vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2204,6 +2204,9 @@ public FooterRow appendFooterRow() {
     public void addThemeVariants(GridVariant... variants) {
         getThemeNames().addAll(Stream.of(variants)
                 .map(GridVariant::getVariantName).collect(Collectors.toList()));
+        if (Stream.of(variants).anyMatch(variant -> variant == GridVariant.LUMO_COMPACT)) {
```

Comment:
This will only work if a developer uses `addThemeVariants` API for the purpose. If you use `grid.getThemeNames().add(GridVariant.LUMO_COMPACT.getVariantName());`, the issue is still there => should be fixed on the Web Component (notifyResize on `theme` -attribute change)

Reply:
Good point, I will close this in favor of that.